pkg/apis/kops/validation/legacy.go

```diff
@@ -731,3 +671,43 @@ func DeepValidate(c *kops.Cluster, groups []*kops.InstanceGroup, strict bool) er
 
 	return nil
 }
+
+func validateKubelet(k *kops.KubeletConfigSpec, c *kops.Cluster, kubeletPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if k != nil {
+
+		{
+			// Flag removed in 1.6
+			if k.APIServers != "" {
+				allErrs = append(allErrs, field.Forbidden(
+					kubeletPath.Child("apiServers"),
+					"api-servers flag was removed in 1.6"))
+			}
+		}
+
+		if c.IsKubernetesGTE("1.10") {
+			// Flag removed in 1.10
+			if k.RequireKubeconfig != nil {
+				allErrs = append(allErrs, field.Forbidden(
+					kubeletPath.Child("requireKubeconfig"),
+					"require-kubeconfig flag was removed in 1.10.  (Please be sure you are not using a cluster config from `kops get cluster --full`)"))
+			}
+		}
+
+		if k.BootstrapKubeconfig != "" {
```

Comment:
It seems that this was only validated on `Spec.Kubelet` but will now be validated on `Spec.MasterKubelet` as well. I think that should be safe but its probably worth double checking.

Reply:
I should have commented on that. Given that any flag added to `kubelet` is also added to `masterKubelet` I figured that this is just the lack of copy/paste. I don't see a reason why you wouldn't want to add the flag if it was only explicitly added to masters, but not when they were added to _both_ workers and masters.